etl-destinations/src/bigquery/client.rs

```diff
@@ -568,7 +568,20 @@ fn bq_error_to_etl_error(err: BQError) -> EtlError {
         BQError::TonicInvalidMetadataValueError(_) => {
             (ErrorKind::ConfigError, "BigQuery invalid metadata value")
         }
-        BQError::TonicStatusError(_) => (ErrorKind::DestinationError, "BigQuery gRPC status error"),
+        BQError::TonicStatusError(status) => {
+            // We have no real clarify from BigQuery when an operation fails because the table is absent
+            // so here we are trying to do best effort.
+            //
+            // From our testing, when trying to send data to a missing table, this is the error that is
+            // returned:
+            // `Status { code: PermissionDenied, message: "Permission 'TABLES_UPDATE_DATA' denied on
+            // resource 'x' (or it may not exist).", source: None }`
+            if status.message().ends_with("(or it may not exist).") {
```

Comment:
Quite unhappy about this, but could not find anything better for now. An alternative would be to actually check for table existence via a query, but it's not great performance-wise.

Reply:
Yeah, that was also a middleground which I was thinking about. We could do that yes, I will just return as destination error like before. Unfortunately the lib doesn't export tonic types so we would have to add it as a dependency to pattern match the status code, I will see what I can do.